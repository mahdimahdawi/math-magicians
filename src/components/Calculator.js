import React, { useState } from 'react';

import calculate from '../logic/calculate';

const UI = () => {
  const [state, setStat] = useState({
    total: 0,
    operation: null,
    next: null,
  });

  const onClickHandler = (e) => {
    const result = calculate(state, e.target.innerHTML);
    setStat(result);
  };

  const { total, operation, next } = state;

  return (
    <div className="calculator">
      <div className="calc">
        <div className="row1">
          <h2 className="result">
            { total }
            { operation }
            { next }
          </h2>
        </div>
        <div className="row">
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            AC
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            +/-
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            %
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper btn-color">
            รท
          </button>
        </div>
        <div className="row">
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            7
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            8
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            9
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper btn-color">
            x
          </button>
        </div>
        <div className="row">
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            4
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            5
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            6
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper btn-color">
            -
          </button>
        </div>
        <div className="row">
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            1
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            2
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            3
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper btn-color">
            +
          </button>
        </div>
        <div className="row">
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper zero">
            0
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper">
            .
          </button>
          <button type="button" onClick={(e) => { onClickHandler(e); }} className="btn-wrapper btn-color">
            =
          </button>
        </div>
      </div>
    </div>
  );
};

export default UI;
