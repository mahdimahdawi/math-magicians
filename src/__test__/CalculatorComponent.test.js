import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('testing Calculator Component', () => {
  test('testing if Calculator Component rendered', () => {
    render(<Calculator />);

    // eslint-disable-next-line testing-library/await-async-query
    expect(screen.findByText('Lets Do Math!')).toMatchSnapshot();
  });

  test('testing AC button click', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByText('AC'));

    expect(screen.getByText('AC')).toMatchInlineSnapshot(`
<button
  class="btn-wrapper"
  type="button"
>
  AC
</button>
`);
  });

  test('testing 6 button click', () => {
    render(<Calculator />);

    fireEvent.click(screen.getByText('AC'));

    expect(screen.getByText('6')).toMatchInlineSnapshot(`
<button
  class="btn-wrapper"
  type="button"
>
  6
</button>
`);
  });
});