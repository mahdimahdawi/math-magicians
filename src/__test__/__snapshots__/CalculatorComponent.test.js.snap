// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Calculator Component testing if Calculator Component rendered 1`] = `Promise {}`;
